packages/flutter/test/material/tabs_test.dart

```diff
@@ -2350,4 +2350,72 @@ void main() {
     expect(find.text('Tab1'), findsOneWidget);
     expect(find.text('Tab2'), findsOneWidget);
   });
+  testWidgets('DefaultTabController should allow for a length of zero', (
+      WidgetTester tester) async {
+    // Test for https://github.com/flutter/flutter/issues/37139
+    final List<Widget> tabList = <Widget>[];
+    final List<Widget> tabViewList = <Widget>[];
+
+    await tester.pumpWidget(
+      MaterialApp(
+        home: StatefulBuilder(
+          builder: (BuildContext context, StateSetter setState) {
+            return DefaultTabController(
+                length: tabList.length,
+                child: Scaffold(
+                  appBar: AppBar(
+                    title: const Text('Default TabBar Preview'),
+                    bottom: (tabList.isNotEmpty)
+                        ? TabBar(isScrollable: true, tabs: tabList)
+                        : null,
+                  ),
+                  body: (tabList.isNotEmpty)
+                      ? TabBarView(children: tabViewList)
+                      : const Center(
+                    child: Text('No tabs'),
+                  ),
+                  bottomNavigationBar: BottomAppBar(
+                    child: Row(
+                      mainAxisSize: MainAxisSize.max,
+                      mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                      children: <Widget>[
+                        IconButton(
+                          key: const Key('Add tab'),
+                          icon: const Icon(Icons.add),
+                          onPressed: () {
+                            setState(() {
+                              tabList.add(
+                                  Tab(text: 'Tab ${tabList.length + 1}'));
+                              tabViewList.add(Center(child: Text(
+                                  'Tab ${tabViewList.length + 1}')));
+                            });
+                          },
+                        ),
+                        IconButton(
+                          key: const Key('Del tab'),
+                          icon: const Icon(Icons.delete),
+                          onPressed: () {
+                            setState(() {
+                              tabList.removeLast();
+                              tabViewList.removeLast();
+                            });
+                          },
+                        )
+                      ],
+                    ),
+                  ),
+                ));
+          },
+        ),
+      ),
+    );
+    expect(find.text('No tabs'), findsOneWidget);
+    await tester.tap(find.byKey(const Key('Add tab')));
+    await tester.pumpAndSettle();
+    expect(find.text('Tab 1'), findsNWidgets(2));
```

Comment:
I modified the `TabBarView`'s text in my example to allow for checking both in isolation, since that is probably much more readable to future maintainers. So, check for "Tab {number}'s view" in `TabBarView`, and also check for "Tab {number}" in `TabBar`